src/app/contact/contact.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { ContactService } from '../core/ContactService';
+import { Message } from '../models/Message';
+
+@Component({
+  selector: 'app-contact',
+  templateUrl: './contact.component.html',
+  styleUrls: ['./contact.component.scss'],
+})
+export class ContactComponent implements OnInit {
+  contactForm = new FormGroup({
+    email: new FormControl('', [
+      Validators.required,
+      Validators.pattern('^[a-z0-9._%+-]+@[a-z0-9.-]+.[a-z]{2,4}$'),
+    ]),
+    message: new FormControl('', [
+      Validators.required,
+      Validators.minLength(5),
+      Validators.maxLength(200),
+    ]),
+  });
+
+  showSummary: boolean;
+
+  constructor(private contactService: ContactService) {}
+
+  ngOnInit(): void {}
+
+  onSubmit() {
+    this.contactService.sendMessage(
+      new Message(this.email.value, this.message.value)
```

Comment:
fuuuuj :D 
nie robimy tak w TypeScripcie  :)
najłatwiej zrobic interfejs Message a pozniej tylko stworzyc zmienna albo obiekt { } który implementuje to co zakłąda interejs. Tworzenie do tego instancji klasy nie jest potrzebne.

Druga sprawa - co jesli nei uda sie wysłanie formularza? :) bo np ktos nie ma internetu? WYswietlisz podziekowanie ale wiadomosci w ogole nie pojdzie. Takie wiadomosci warto wyswietlac po zakonczeniu requestu http

Reply:
Poprawione obie rzeczy.